api/agent/lb_agent.go

```diff
@@ -258,5 +300,18 @@ func (a *lbAgent) handleCallEnd(ctx context.Context, call *call, err error, isFo
 	return err
 }
 
+func (w *DetachedResponseWriter) Header() http.Header {
+	return w.origin.Header()
+}
+
+func (w *DetachedResponseWriter) Write(data []byte) (int, error) {
+	return w.origin.Write(data)
+}
+
+func (w *DetachedResponseWriter) WriteHeader(statusCode int) {
+	w.origin.WriteHeader(statusCode)
+	w.acked <- struct{}{}
```

Comment:
is this correct? we want to ack immediately after placement but before the task actually runs, won't this only be acked once the task has begun to write the response (in theory, near the end of the task)?

Reply:
this change adds a new call listener and if the call is detached in the BeforeCall method we send a RunnerMsg_ResultStart which will trigger the writing of the header on the lb_agent via the runner_client, so in detached mode we send an ack during the BeforeCall